### ISSUE TITLE
GPL license prevents template usage in non-GPL projects

### ISSUE THREAD

**Issue by DrTobe** [AdinAck/cargo-embassy#24]
@embediver and I, we wanted to use `cargo-embassy` to bootstrap an embassy based project which we would probably like to release under a non-copyleft license. If I understand it correctly, `cargo-embassy` copies the template files from `src/templates` into the generated project. In my understanding, these template files are also licensed under GPL-3.0 then, so the copyleft-nature of the GPL would affect all generated projects which would probably need to be licensed under GPL-3.0, too.

Since most projects in the Rust world seem to be `MIT or Apache-2.0`, I am not sure if this is what you intended. If you only wanted to license the non-template files under GPL but allow to license the generated projects under whatever license the users want, this needs clarification, as far as I understand.

**Comment by AdinAck** [comment]
I agree that the generated output of `cargo-embassy` falls under the GPLv3 license as it is now. I don't know how to change this, but am willing to.

**Comment by embediver** [comment]
We had a internal discussion about this and came to the conclusion, that it should be enough to explicitly state that no copyright is held for the generated code and structure (from a user standpoint, the code is generated and in the public domain, as far as our understanding goes).
Also it is debatable, if the threshold of originality for the generated code is high enough for a license to apply.

A statement might look like this:

_"Note that no copyright is held for the generated project structure by the cargo-embassy authors. The generated project may be freely used, without restrictions, and you may freely put it under any license of your choice."_

A lawyer might formulate parts differently, but this should be sufficient.

**Comment by 3210jr** [comment]
@AdinAck - thanks for the awesome project! Just pinging this response from @embediver to include this note in the README. Could help save a lot of time for devs looking to control the licensing of their products.
